internal/cloudapi/v2/v2_test.go

```diff
@@ -814,6 +794,7 @@ func TestComposeJobError(t *testing.T) {
 }
 
 func TestComposeCustomizations(t *testing.T) {
+	// Test is flaky if we use t.TempDir()
```

Comment:
This is weird, do you have by any chance a test log?

Reply:
Yes. The first build failed https://copr.fedorainfracloud.org/coprs/packit/osbuild-osbuild-composer-2417/build/3652686/ and I had to revert 

1. centos-stream-9-aarch64: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/centos-stream-9-aarch64/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:967: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations4269165628/001: directory not empty
    ```

1. fedora-34-aarch64: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-34-aarch64/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestImageTypes (0.02s)
        testing.go:958: TempDir RemoveAll cleanup: unlinkat /tmp/TestImageTypes501896041/001: directory not empty
    ```

1. fedora-35-aarch64: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-35-aarch64/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestCompose (0.02s)
        testing.go:958: TempDir RemoveAll cleanup: unlinkat /tmp/TestCompose2164754594/001: directory not empty

    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:958: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations1236627265/001: directory not empty
    ```

1. fedora-35-ppc64le: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-35-ppc64le/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:1091: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations1419833003/001: directory not empty
    ```

1. fedora-36-aarch64: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-36-aarch64/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:1091: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations3910835978/001: directory not empty
    ```

1. fedora-36-s390x: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-36-s390x/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:1091: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations3910835978/001: directory not empty
    ```

1. fedora-rawhide-aarch64: [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/packit/osbuild-osbuild-composer-2417/fedora-rawhide-aarch64/03652686-osbuild-composer/builder-live.log.gz)
    ```
    --- FAIL: TestComposeCustomizations (0.00s)
        testing.go:1090: TempDir RemoveAll cleanup: unlinkat /tmp/TestComposeCustomizations2466484498/001: directory not empty
    ```

Reply:
Ah, this goroutine (https://github.com/osbuild/osbuild-composer/blob/761aab6cac9c9528a02e1ce479ef9747556ed525/internal/cloudapi/v2/v2_test.go#L46) probably writes something into the temporary directory after it's removed. See https://github.com/golang/go/issues/43547 More synchronization would be needed to fix this properly.

Thinking of this, it can be an issue in all tests that call `newV2Server`. Can you please revert the change for all tests that use `newV2Server`?

Reply:
Reverted all tests that use `newV2Server` in 8302a33.